great_expectations/data_context/types/resource_identifiers.py

```diff
@@ -225,14 +229,23 @@ def ge_cloud_id(self):
     def ge_cloud_id(self, value) -> None:
         self._ge_cloud_id = value
 
+    @property
+    def resource_name(self) -> str:
+        return self._resource_name
+
     def to_tuple(self):
-        return (self.resource_type, self.ge_cloud_id)
+        return (self.resource_type, self.ge_cloud_id, self.resource_name)
```

Comment:
So this is the only key type that is length 3? How can generalize some of these tuple conversions so we don't have to `if len x ... else ...`?

Reply:
```python
return tuple(x for x in (self.resource_type, self.ge_cloud_id, self.resource_name) if x )
```

Reply:
Could also make it return `namedtuple` with an optional `resource_name`
https://docs.python.org/3/library/typing.html#typing.NamedTuple

Reply:
Totally agree this should not be represented as a tuple. I'd like to revisit that effort (refactoring our stores to not use tuples as keys) in a separate PR if that works for ya'll.